frontend/README.md

```diff
@@ -142,6 +142,45 @@ it('does something', () => {
 We use `jest`'s [`expect`](https://facebook.github.io/jest/docs/en/expect.html) assertion tool.
 
 
+## Localization
+
+The user interface is localized using [Fluent](https://projectfluent.org/) and the library `fluent-react`. Fluent allows to move the complexity of handling translated content from the developer to the translator. Thus, when using it, you should care only about the English version, and trust that the localizers will know what to do with your string in their language.
+
+### Adding localized content
+
+Every time you add a piece of UI that contains some content, there are 2 things you need to do. First, make sure you put it inside a `<Localized>` element, and give it an id. String identifiers should follow this form: `${module}-${component}-${contentDescription}`. For example, if you were to add a new button (`<button>Update</button>`) to the `Editor` component in the `entitydetails` module, the id for that button might be `entitydetails-editor-button-update`.
+
+That would give:
+
+```js
+class Editor extends React.Component {
+    render() {
+        return <div>
+            <Localized id="entitydetails-editor-button-update">
+                <button>Update</button>
+            </Localized>
+        </div>;
+    }
+}
+```
+
+The second thing you need to do is to add that new string into our main translation files. Translations available to localizers are based on those files, so if you forget to put the new content there, no one will be able to translate that content.
+
+The source language for our application is `en-US` (that is what you should use by default when writing content in the code, and in the language files). You can find them in `frontend/public/static/locale/en-US/`. We currently have only one file, named `translate.ftl`, and that is where all content should be added.
+
+Those files use the FTL format. In its simplest form, a string in such a file (called a `message` in Fluent vocabulary) looks like this: `message-id = Content`. There are, however, lots of tools that you can use in that syntax, and you are encouraged to read [Fluent's Syntax Guide](https://projectfluent.org/fluent/guide/) to familiarize yourself with them. Make sure you also take a look at the [Good Practices for Developers](https://github.com/projectfluent/fluent/wiki/Good-Practices-for-Developers#prefer-separate-messages-over-variants-for-ui-logic) guide.
+
+### Social contract
+
+Fluent uses the concept of a *social contract* between developer and localizers. This contract is established by the selection of a unique identifier, called `l10n-id`, which carries a promise of being used in a particular place to carry a particular meaning.
```

Comment:
Some of use this phrase, but I'm not happy with _Social Contract_. This is a development pattern, and an API surface to external code and processes that is needed to get continuous localization and change management in general.

I prefer to word this around semantic versioning for strings, with no "middle" version number. If you're breaking backwards compat, use a new ID. If you're not, seriously, don't use a new ID.

I also like to describe the l10n ids as good variable names. Often, if you change the semantics around a given UI, a different name for that variable is part of the change.